### ISSUE TITLE
Series upgraded but SZ do not download new Subtitles.

### ISSUE THREAD

**Issue by cphmichael** [pannal/Sub-Zero.bundle#670]
Unfortunately, it seems like there still is a problem with downloading subtitles.

Yesterday evening from Thu Sep 05 20:35 – Fri Sep 06 00:33, Sonarr upgraded The Blacklist S06E01 – S06E22 from WEBDL to Bluray with included PGS subtitles.

I was very happy to see that when the first “The Blacklist S6E01” was upgraded, SZ (auto) downloaded a new SRT External subtitles.

For the subsequently S06E02-S06E22, SZ did not download subtitles.

In Kitana I did a “Searching for English subs”.

As you can see in the attached screen dump, there are available subtitles with score over 340, which is my minimum score for TV.

[com.plexapp.agents.subzero.0.log](https://github.com/pannal/Sub-Zero.bundle/files/3582732/com.plexapp.agents.subzero.0.log)

[com.plexapp.agents.subzero.1.log](https://github.com/pannal/Sub-Zero.bundle/files/3582734/com.plexapp.agents.subzero.1.log)

[com.plexapp.agents.subzero.2.log](https://github.com/pannal/Sub-Zero.bundle/files/3582736/com.plexapp.agents.subzero.2.log)

[com.plexapp.agents.subzero.3.log](https://github.com/pannal/Sub-Zero.bundle/files/3582737/com.plexapp.agents.subzero.3.log)

![PMS](https://user-images.githubusercontent.com/45288008/64409541-67ad0500-d089-11e9-9425-4eac9d1f2521.PNG)

![Searching for English subs](https://user-images.githubusercontent.com/45288008/64409553-6c71b900-d089-11e9-8d85-c2881ebc39a2.PNG)

**Comment by pannal** [comment]
Yes, this might still be an issue with updating when upgrading videos.
Plex doesn't seem to always trigger full refreshes on updates - you can disable scanning in Plex and let Sonarr trigger Plex on update (in the Connect tab).

I'm not 100% sure that this will lead to an actual full refresh, but feel free to try.

**Comment by pannal** [comment]
Hmm wait. On the upgrade in `com.plexapp.agents.subzero.3.log` S06E01 has a filename of `The Blacklist - S06E01 - Dr. Hans Koehler Bluray-1080p.mkv` - was that `The Blacklist - S06E01 - Dr. Hans Koehler.mkv` before?

If those don't change on upgrade, it might not trigger a re-search.

Although, `The Blacklist - S06E03 - The Pharmacist.mkv` had a subtitle with the name `The Blacklist - S06E03 - The Pharmacist.en.srt`. Was that the wrong one?

**Comment by cphmichael** [comment]
In PMS I had ticked all the library scanning.

I have now un-ticked:
* Scan my library automatically
* Run a partial scan when changes are detected
* Include music libraries in automatic updates
* Scan my library periodically

In Sonarr I already had setup a PMS connect with Update Library set to Yes.

In Sonarr I have "Rename Episodes" set to Yes

If I in Sonarr, Media Management set Advanced Settings = Show, I can see that "Standard Episode Format" has the {Quality Full}

I think that is what linuxserver Sonarr came with when I installed it.

I changed Sonarr from binhex to linuxserver a while ago. Maybe binhex Sonarr didn't have the  {Quality Full}

I had a problem with binhex NZBGet. When I received very big downloads, binhex NZBGet hanged (stalled) in the process after downloading the files.

I changed to linuxserver NZBGet that seems not to have this problem. I therefore also changed to linuxserver Sonarr and Radarr.

Do you want me to remove the {Quality Full} or I can set "Rename Episodes = No" if that is better?

![PMS library scan](https://user-images.githubusercontent.com/45288008/64479129-c621d280-d1b2-11e9-84b7-9938bc2da6ff.png)

![Sonarr Update Library](https://user-images.githubusercontent.com/45288008/64479137-cd48e080-d1b2-11e9-9032-952f81b88e9b.png)

![Sonarr rename](https://user-images.githubusercontent.com/45288008/64479139-d20d9480-d1b2-11e9-93eb-743f3eb05075.png)

**Comment by pannal** [comment]
I don't really get what the issue is, here. You've mentioned `For the subsequently S06E02-S06E22, SZ did not download subtitles.` - SZ found subtitles on disk for those episodes, though, judging by the logs and thus didn't download any.

**Comment by cphmichael** [comment]
As you can see on the attached screen dump of the Blacklist folder, there is only a few .srt files.

Sonarr send me an email that Blacklist S6E1 was upgraded from WEBDL to Bluray.

I logged on to my Plex client on my Nividia Shield and saw that there was an external .srt subtitles for Blacklist S6E1.

I was happy. The fix you had made in the SZ DEV worked :-)

Later I got other emails from Sonarr, that more Blacklist S6 was upgraded.

After some more Sonarr emails, I logon on again to my Plex client to check the external subtitles.

They were all (S6E2-S6E22) without external subtitles.

I started playing Blacklist S6E1. To check that the external subtitles was the right one. That it match the series. The subtitles was fine.

I played S6E1 for a few minutes. When I moved on to S6E2 the playing of S6E1 had triggered a subtitles download for S6E2. I think because I have ticked:

* On media playback: search for missing subtitles (refresh item) / hybrid-plus: current item and next episode.

I have ticked:

* Automatically delete leftover/unused (externally saved) subtitles

What happens when Sonarr upgrade the Blacklist WEBDL to Bluray?

Is SZ deleting the old external .srt file - and at the same time SZ has to download a new external .srt for the Bluray version.

![Blacklist directory](https://user-images.githubusercontent.com/45288008/64480479-97aef200-d1c8-11e9-92f0-fc835de86b23.png)

![PMS Blacklist S6E4](https://user-images.githubusercontent.com/45288008/64480481-9d0c3c80-d1c8-11e9-9728-c1ca92f054ef.png)

**Comment by pannal** [comment]
SZ auto cleans up after it's done with one "call" to the agent. So yes, it will kill all subtitles that don't have any corresponding video files left, after the first upgrade for example.

I'm not sure what happens here. Upgrades have always been an issue. Please keep an eye on this, as I can't figure out from your logs what happened.

**Comment by cphmichael** [comment]
You could maybe make a Debug option called 'Debug Extended' with focus on the upgrade issue.
That could help you solve the problem :-)

**Comment by cphmichael** [comment]
Today it worked :-)

Sonarr upgradede Dark Phoenix from Remux-1080p to Remux-2160p and SZ downloaded a new subtitles.

Maybe the un-tick of the PMS library scans solved the problem.

[com.plexapp.agents.subzero.log](https://github.com/pannal/Sub-Zero.bundle/files/3590831/com.plexapp.agents.subzero.log)

<img width="645" alt="sonarr" src="https://user-images.githubusercontent.com/45288008/64536913-3008bc00-d31a-11e9-8418-29332502d0fa.PNG">

<img width="561" alt="kitana" src="https://user-images.githubusercontent.com/45288008/64536923-34cd7000-d31a-11e9-9eb8-d740346a4c87.PNG">

<img width="356" alt="folder" src="https://user-images.githubusercontent.com/45288008/64537125-84ac3700-d31a-11e9-8f5d-ed55de7e8407.PNG">

**Comment by pannal** [comment]
Can it be that this issue is only related to TV Shows?

**Comment by felho001** [comment]
Hi i coudnt find this advanced menu plex and subzero are last version.

and this setup no show up:

https://github.com/pannal/Sub-Zero.bundle/wiki/User-Guide

**Comment by pannal** [comment]
@felho001 please don't highjack others' tickets. 
Your answer is in #655

**Comment by cphmichael** [comment]
I haven't had any movie/TV-show upgrade for a while, so I do not know I if is related only to TV-Shows.

Maybe the un-tick of the PMS library scans solved the problem, but I need more "upgrade" before I can answer that.
